src/main.rs

```diff
@@ -1,14 +1,38 @@
 extern crate libc;
 use libc::c_char;
 use libc::c_int;
+use libc::c_void;
 use std::ffi::CStr;
 use std::ffi::CString;
+use std::ptr;
+
+#[repr(C)]
+struct deno_buf {
+    data: *const c_char,
+    len: c_int, // TODO(ry) should be size_t.
```

Comment:
In JS, (typed)arrays can't be longer than UINT32_MAX. So maybe this is appropriate.